### ISSUE TITLE
Extend return format

### ISSUE THREAD

**Issue by bramus** [bramus/js-specificity#2]
Instead of returning a simple Object with a,b,c keys, return a more extended object (or new datatype?) that contains more info. See the return format from https://github.com/keeganstreet/specificity for inspiration:

```js
{
    selector: 'ul#nav li.active a',
    specificity: '0,1,1,3',
    specificityArray: [0, 1, 1, 3],
    parts: [
      { selector: 'ul', type: 'c', index: 0, length: 2 },
      { selector: '#nav', type: 'a', index: 2, length: 4 },
      { selector: 'li', type: 'c', index: 5, length: 2 },
      { selector: '.active', type: 'b', index: 8, length: 7 },
      { selector: 'a', type: 'c', index: 13, length: 1 }
    ]
}
```

**Comment by bramus** [comment]
Currently, I see several roads that can be taken.

1. Return an extended Object:
    
    ```}
    return {
        selector: generate(selector),
        asObject: specificity,
        asArray: [specificity.a, specificity.b, specificity.c],
        asString: `(${specificity.a},${specificity.b},${specificity.c})`,
    };
    ```
    
1. Return an instance of a Class:

    ```js
    return new Specificity(specificity, selector);
    ```

    The class would be this:
    
    ```js
    class Specificity {
        constructor(values, selector) {
            this.values = values;
            this.selectorObject = selector;
        }

        selector() {
            return generate(this.selectorObject);
        }

        toObject() {
            return this.values;
        }

        toArray() {
            return [this.values.a, this.values.b, this.values.c];
        }

        toString() {
            return `(${this.values.a},${this.values.b},${this.values.c})`;
        }

        toJSON() {
            return {
                selector: this.selector(),
                asObject: this.toObject(),
                asArray: this.toArray(),
                asString: this.toString(),
            };
        }
    }
    ```
    
1. Return a simple Object like before (i.e. `{a:0, b:0, c:0}`), but also provide a conversion function that can convert between an array, a string, etc.

I'm leaning towards option 2 right now …

**Comment by bramus** [comment]
Currently, I see several roads that can be taken.

1. Return an extended Object:
    
    ```}
    return {
        selector: generate(selector),
        asObject: specificity,
        asArray: [specificity.a, specificity.b, specificity.c],
        asString: `(${specificity.a},${specificity.b},${specificity.c})`,
    };
    ```
    
1. Return an instance of a Class:

    ```js
    return new Specificity(specificity, selector);
    ```

    The class would be this:
    
    ```js
    class Specificity {
        constructor(values, selector) {
            this.values = values;
            this.selectorObject = selector;
        }

        selector() {
            return generate(this.selectorObject);
        }

        toObject() {
            return this.values;
        }

        toArray() {
            return [this.values.a, this.values.b, this.values.c];
        }

        toString() {
            return `(${this.values.a},${this.values.b},${this.values.c})`;
        }

        toJSON() {
            return {
                selector: this.selector(),
                asObject: this.toObject(),
                asArray: this.toArray(),
                asString: this.toString(),
            };
        }
    }
    ```
    
1. Return a simple Object like before (i.e. `{a:0, b:0, c:0}`), but also provide a conversion function that can convert between an array, a string, etc.

I'm leaning towards option 2 right now …

**Comment by bramus** [comment]
Currently, I see several roads that can be taken.

1. Return an extended Object:
    
    ```}
    return {
        selector: generate(selector),
        asObject: specificity,
        asArray: [specificity.a, specificity.b, specificity.c],
        asString: `(${specificity.a},${specificity.b},${specificity.c})`,
    };
    ```
    
1. Return an instance of a Class:

    ```js
    return new Specificity(specificity, selector);
    ```

    The class would be this:
    
    ```js
    class Specificity {
        constructor(values, selector) {
            this.values = values;
            this.selectorObject = selector;
        }

        selector() {
            return generate(this.selectorObject);
        }

        toObject() {
            return this.values;
        }

        toArray() {
            return [this.values.a, this.values.b, this.values.c];
        }

        toString() {
            return `(${this.values.a},${this.values.b},${this.values.c})`;
        }

        toJSON() {
            return {
                selector: this.selector(),
                asObject: this.toObject(),
                asArray: this.toArray(),
                asString: this.toString(),
            };
        }
    }
    ```
    
1. Return a simple Object like before (i.e. `{a:0, b:0, c:0}`), but also provide a conversion function that can convert between an array, a string, etc.

I'm leaning towards option 2 right now …

**Comment by bramus** [comment]
This change will be included in `2.0.0`. You can try out a preview of this release by installing `@bramus/specificity@next`. See [the updated README](https://github.com/bramus/js-specificity/blob/fb965b5f88ab1ee7ab9b61a1235ab84423b802da/README.md) on how to use.
